libc/string_lenchr.c

```diff
@@ -281,37 +281,52 @@ TEST(string_chr, ascii)
 }
 
 
-TEST(string_chr, not_ascii)
+TEST(string_chr, not_ascii_chars)
 {
-	const char notAsciiString[] = "♥♣♠◊⊗こんにちは❉❉⌨⌨⌨⌨⌨⌨⌨⌨❉❉♦x";
-	char notAsciiSet[129];
+	char notAsciiSet[129] = { 0 };
 	int sz, i;
 
-	/* Checking ability to read the chars out of ascii charset */
-	sz = sizeof(notAsciiString) - 1;
-	TEST_ASSERT_EQUAL_STRING("x", strchr(notAsciiString, 'x'));
-	TEST_ASSERT_EQUAL_STRING("x", strrchr(notAsciiString, 'x'));
-	TEST_ASSERT_EQUAL_STRING("x", memchr(notAsciiString, 'x', sz));
+	/*
+	 * Depending on architecture we will get
+	 * output {128;255} if chars are unsigned or {-128;-1} when chars are signed
+	 */
 
-	/* Checking  out of ascii bytes */
+	notAsciiSet[128] = 0;
+	sz = sizeof(notAsciiSet);
 	for (i = 128; i <= 255; i++) {
 		notAsciiSet[i - 128] = i;
+		/* Testing capability of functions to hold and read not ascii set */
+		TEST_ASSERT_EQUAL_STRING(&notAsciiSet[i - 128], strchr(notAsciiSet, notAsciiSet[i - 128]));
+		TEST_ASSERT_EQUAL_STRING(&notAsciiSet[i - 128], strrchr(notAsciiSet, notAsciiSet[i - 128]));
+		TEST_ASSERT_EQUAL_STRING(&notAsciiSet[i - 128], memchr(notAsciiSet, notAsciiSet[i - 128], sz));
```

Comment:
Small simplification.
```suggestion
	notAsciiSet[128] = 0;
	sz = sizeof(notAsciiSet);
	for (i = 0; i <= 127; i++) {
		notAsciiSet[i] = i + 128;
		/* Testing capability of functions to hold and read not ascii set */
		TEST_ASSERT_EQUAL_STRING(&notAsciiSet[i], strchr(notAsciiSet, notAsciiSet[i]));
		TEST_ASSERT_EQUAL_STRING(&notAsciiSet[i], strrchr(notAsciiSet, notAsciiSet[i]));
		TEST_ASSERT_EQUAL_STRING(&notAsciiSet[i], memchr(notAsciiSet, notAsciiSet[i], sz));
```